From 544ca207a5dc481ad4f95ad378f552414cb709b9 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Wed, 19 Jan 2011 20:04:35 +0800
Subject: [PATCH 10/38] ENGR00138198 rild: fix Amazon modem can't pppd bug.

1. Amazon modem have an new product ID. Add this in runtime check.
2. fix a response cause rild SEGENV bug.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 reference-ril/reference-ril.c | 9 +--------
 reference-ril/runtime_port.c  | 8 ++++++++
 2 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/reference-ril/reference-ril.c b/reference-ril/reference-ril.c
index 81a4819..3f5682a 100755
--- a/reference-ril/reference-ril.c
+++ b/reference-ril/reference-ril.c
@@ -1671,14 +1671,7 @@ static void requestSetupDataCall(void *data, size_t datalen, RIL_Token t)
     char *cmd;
     int err;
     ATResponse *p_response = NULL;
-    char *response2[2] = { "1", PPP_TTY_PATH };
-    char *response3[3] = { "1", PPP_TTY_PATH, ""};
-    char **response;
-
-    if (current_modem_type == HUAWEI_MODEM)
-	    response = response3;
-    else
-	    response = response2;
+    char *response[3] = { "1", PPP_TTY_PATH, ""};
 
     apn = ((const char **)data)[2];
 
diff --git a/reference-ril/runtime_port.c b/reference-ril/runtime_port.c
index 2ce32ab..0f4758d 100644
--- a/reference-ril/runtime_port.c
+++ b/reference-ril/runtime_port.c
@@ -84,6 +84,14 @@ static struct modem_3g_device modem_3g_device_table[] = {
 		.deviceport	= "/dev/ttyACM3",
 		.dataport	= "/dev/ttyACM0",
 		.type		= AMAZON_MODEM,
+	},
+	{
+		.name		= "InnoComm-Amazon1",
+		.idVendor	= "1519",
+		.idProduct	= "0020",
+		.deviceport	= "/dev/ttyACM3",
+		.dataport	= "/dev/ttyACM0",
+		.type		= AMAZON_MODEM,
 	}
 };
 
-- 
1.8.0

